File: modules/chat-core/src/main/java/com/shterneregen/securelan/chat/service/ChatBroadcastService.java
package com.shterneregen.securelan.chat.service;

import com.shterneregen.securelan.chat.transport.ChatSocketSession;

public interface ChatBroadcastService {
    void addClient(String nickname, ChatSocketSession session);
    void removeClient(String nickname);
    void syncPeers(ChatSocketSession session, String excludeNickname);
    void publishUserJoined(String nickname);
    void publishUserLeft(String nickname);
    void publishMessage(String senderNickname, String text);
    void publishSignal(String senderNickname, String payload);
}

File: modules/chat-core/src/main/kotlin/com/shterneregen/securelan/chat/service/ChatBroadcastService.kt
package com.shterneregen.securelan.chat.service

import com.shterneregen.securelan.chat.transport.ChatSocketSession

interface ChatBroadcastService {
    fun addClient(nickname: String, session: ChatSocketSession)
    fun removeClient(nickname: String)
    fun syncPeers(session: ChatSocketSession, excludeNickname: String?)
    fun publishUserJoined(nickname: String)
    fun publishUserLeft(nickname: String)
    fun publishMessage(senderNickname: String, text: String)
    fun publishSignal(senderNickname: String, payload: String)
}
